subsys/nrf_security/src/ssf_secdom/sdfw_psa_ns_api.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/cache.h>
+
+#include <psa/client.h>
+#include <psa/error.h>
+
+#include <tfm/sdfw_psa_ipc_service.h>
+
+#include "sdfw_psa_ns_ipc.h"
+#include "bounce_buffers.h"
+
+/* The correctness of the serialization depends on these asserts */
+BUILD_ASSERT(4 == sizeof(psa_handle_t));
+BUILD_ASSERT(4 == sizeof(int32_t));
+BUILD_ASSERT(4 == sizeof(const psa_invec *));
+BUILD_ASSERT(4 == sizeof(size_t));
+BUILD_ASSERT(4 == sizeof(psa_outvec *));
+BUILD_ASSERT(4 == sizeof(psa_status_t *));
+
+psa_status_t psa_call_buffered_and_flushed(psa_handle_t handle, int32_t type,
+					   const psa_invec *in_vec, size_t in_len,
+					   psa_outvec *out_vec, size_t out_len)
+{
+	/* We have no need for this at this time */
+	ARG_UNUSED(type);
+
+	psa_status_t ipc_status = sdfw_psa_ns_ipc_setup();
+
+	if (ipc_status != PSA_SUCCESS) {
+		return ipc_status;
+	}
+
+	/* volatile and flushed because the cpusec core will usually
+	 * modify this variable
+	 */
+	psa_status_t volatile status = PSA_ERROR_COMMUNICATION_FAILURE;
+	sys_cache_data_flush_range((void *)&status, sizeof(status));
+
+	uint32_t ipc_service_buf[SDFW_PSA_IPC_DATA_LEN];
+
+	ipc_service_buf[0] = handle; /* i.e. TFM_CRYPTO_HANDLE defined to 0x40000100U */
```

Comment:
wait this is a constant? Isn't this used to have multiple concurrent multi-part operations?

Reply:
operation->handle is for multi-part operations.

It is unfortunate that TF-M used the word handle for these two different things.

It will be one value for PSA Crypto, and another value for PSA ITS.

Reply:
ah thanks!